src/middleware/errorHandler.js

```diff
@@ -0,0 +1,51 @@
+const Sentry = require('@sentry/node');
+const express = require('@feathersjs/express');
+const decode = require('jwt-decode');
+
+const { requestError } = require('../logger/systemLogger');
+const logger = require('../logger');
+
+const logRequestInfosInErrorCase = (error, req, res, next) => {
+	if (error && !req.url.includes('/authentication')) {
+		let decodedJWT;
+		try {
+			decodedJWT = decode(req.headers.authorization);
+		} catch (err) {
+			decodedJWT = {};
+		}
+
+		requestError(req, (decodedJWT || {}).userId, error);
+	}
+	next(error);
+};
+
+const formatErrors = (showRequestId) => (error, req, res, next) => {
+	if (error) {
+		error.data = showRequestId ? {	// clear data and add requestId
+			requestId: req.headers.requestId,
+		} : {};
+
+		if (error.stack) { // other errors
+			logger.info(error.stack, showRequestId ? { requestId: req.headers.requestId } : {});
+			delete error.stack;
+		}
+	}
+	next(error);
+};
+
+const returnAsJson = express.errorHandler({
+	html: (error, req, res) => {
+		res.json(error);
+	},
+});
+
+const errorHandler = (app) => {
+	if (process.env.NODE_ENV !== 'test') {
+		app.use(logRequestInfosInErrorCase);
+	}
+	app.use(Sentry.Handlers.errorHandler());
+	app.use(formatErrors(process.env.NODE_ENV !== 'test'));
```

Comment:
separate log_level

Reply:
It is handle that requestId is not return by test, otherwise we must change some tests.